### PR TITLE
Unify tests for `plot_param_importances`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Simplification of visualisation module. 

## Description of the changes

By following #3701

- Remove matplotlib test
- Refactor the plotly test to check retuned values of `_get_param_importances_info` and add matplotlib test

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@contramundum53 Could you review this PR?

File: tests/visualization_tests/test_param_importances.py [hunk 1]
```diff
@@ -1,87 +1,141 @@
...
+
+
+@parametrize_plot_param_importances
+def test_plot_param_importances_customized_target_name(
+    plot_param_importances: Callable[..., Any]
+) -> None:
+
+    params = ["param_a", "param_b"]
+    study = prepare_study_with_trials()
+    figure = plot_param_importances(study, params=params, target_name="Target Name")
```
Comment (contramundum53) [inline tests/visualization_tests/test_param_importances.py:73]: Check save figure?

**Comment by HideakiImamura** [comment]
Note that https://github.com/optuna/optuna/pull/3765 should be merged at first.

**Comment by contramundum53** [comment]
#3765 seems to be merged.

File: tests/visualization_tests/test_param_importances.py [hunk 2]
```diff
@@ -9,76 +13,128 @@
...
-from optuna.visualization import plot_param_importances
+from optuna.visualization import plot_param_importances as plotly_plot_param_importances
+from optuna.visualization._param_importances import _get_importances_info
+from optuna.visualization._plotly_imports import _imports as plotly_imports
+from optuna.visualization.matplotlib import plot_param_importances as plt_plot_param_importances
+from optuna.visualization.matplotlib._matplotlib_imports import _imports as plt_imports
 
 
-def test_target_is_none_and_study_is_multi_obj() -> None:
+if plotly_imports.is_successful():
```
Comment (HideakiImamura) [inline tests/visualization_tests/test_param_importances.py:28]: We don't need this `if` statement.

File: tests/visualization_tests/test_param_importances.py [hunk 3]
```diff
@@ -9,76 +13,128 @@
...
+    from optuna.visualization._plotly_imports import go
 
-    study = create_study(directions=["minimize", "minimize"])
-    with pytest.raises(ValueError):
-        plot_param_importances(study)
 
+if plt_imports.is_successful():
```
Comment (HideakiImamura) [inline tests/visualization_tests/test_param_importances.py:32]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the update. Almost LGTM! I haver two minor comments. PTAL.

File: tests/visualization_tests/test_param_importances.py (hunk 2 above)
Comment (nzw0301) [reply tests/visualization_tests/test_param_importances.py:28]: Definitelly. Thanks!

**Comment by nzw0301** [comment]
@HideakiImamura Thank you for your suggestions. I've fixed them.

File: tests/visualization_tests/test_param_importances.py [hunk 4]
```diff
@@ -9,76 +13,121 @@
...
-
-    # Test with a customized target name.
-    figure = plot_param_importances(study, target_name="Target Name")
-    assert figure.layout.xaxis.title.text == "Importance for Target Name"
+    study.optimize(lambda t: t.suggest_float("x", 0, 1), n_trials=1)
 
-    # Test with wrong parameters.
     with pytest.raises(ValueError):
-        plot_param_importances(study, params=["optuna"])
+        _get_importances_info(study, None, None, None, "Objective Value")
```
Comment (contramundum53) [inline tests/visualization_tests/test_param_importances.py:55]: This is covered in the tests of `get_param_importances`, so I think we don't need to check it here.

**Comment by nzw0301** [comment]
Note that I will not be able to work on this PR (more specifically, the whole Optuna development) until the end of this month at least. So please feel free to take over this PR if my response is not fast enough for the optuna v3 release schedule).

File: tests/visualization_tests/test_param_importances.py [hunk 5]
```diff
@@ -9,76 +13,121 @@
...
+    ],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
```
Comment (contramundum53) [inline tests/visualization_tests/test_param_importances.py:88]: I don't think these cases are necessary.
```suggestion
```

File: tests/visualization_tests/test_param_importances.py [hunk 6]
```diff
@@ -9,76 +13,121 @@
...
+    [create_study, _create_study_with_failed_trial],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
```
Comment (contramundum53) [inline tests/visualization_tests/test_param_importances.py:116]:
```suggestion
```
Ditto.

File: tests/visualization_tests/test_param_importances.py [hunk 7]
```diff
@@ -9,76 +13,121 @@
...
+    specific_create_study: Callable[[], Study], params: Optional[List[str]]
+) -> None:
+
+    study = specific_create_study()
+    info = _get_importances_info(
+        study, None, params=params, target=None, target_name="Objective Value"
+    )
+    assert len(info.importance_values) == 0
+    assert len(info.param_names) == 0
+    assert len(info.importance_labels) == 0
```
Comment (contramundum53) [inline tests/visualization_tests/test_param_importances.py:130]: How about comparing it directly with an `_ImportancesInfo`?
```suggestion
    assert info == _ImportancesInfo(importance_values=[], param_names=[], importance_labels=[], target_name="Objective Value")
```

**Comment by contramundum53** [comment]
Almost LGTM. I left some minor comments above.

**Comment by HideakiImamura** [comment]
@nzw0301 Could you update the PR to resolve the conflict?

**Comment by nzw0301** [comment]
Sure, within 24h.

File: tests/visualization_tests/test_param_importances.py (hunk 4 above)
Comment (nzw0301) [reply tests/visualization_tests/test_param_importances.py]: Right. Thanks.

**Comment by nzw0301** [comment]
@contramundum53 Thank you for your constructive suggestions. I've updated the PR by introducing all suggested changes as in cf393b9.

**Comment by contramundum53** [comment]
@nzw0301 Sorry for being late. Could you also apply https://github.com/optuna/optuna/pull/3760#discussion_r925437575 ?

**Comment by nzw0301** [comment]
@contramundum53 Thank you for your reminder! I've updated PR.

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Comment by contramundum53** [comment]
Thanks for the PR!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**